libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -3472,7 +3476,9 @@ CK_DECLARE_FUNCTION( CK_RV, C_Sign )( CK_SESSION_HANDLE xSession,
 
                     if( lMbedTLSResult != CKR_OK )
                     {
-                        PKCS11_PRINT( ( "mbedTLS sign failed with error %d \r\n", lMbedTLSResult ) );
+                        PKCS11_PRINT( ( "mbedTLS sign failed with error %s : %s \r\n",
+                                        mbedtls_strerror_highlevel( lMbedTLSResult ),
```

Comment:
It the function returns NULL this is technically UB, but most implementations will just print (NULL).

I did quick test on the Xtensa compiler and this hardfaults if NULL is returned.

Reply:
Good point. To address this concern, we could return a default value like "<low-level-code-not-included>"

Reply:
I'm thinking of adding file-specific constants and macros to log level-codes only if non-null. 
```
static const char * pNoHighLevelMbedTlsCodeStr = "<No-High-Level-Code>";
static const char * pNoLowLevelMbedTlsCodeStr = "<No-Low-Level-Code>";

#define mbedTlsHighLevelCodeOrDefault( errnum )                             \
    {                                                                      \
        ( mbedtls_strerror_highlevel( errnum ) != NULL ) ?                  \
        mbedtls_strerror_highlevel( errnum ) : pNoHighLevelMbedTlsCodeStr ); \
    }

#define mbedTlsLowLevelCodeOrDefault( errnum )                             \
    {                                                                      \
        ( mbedtls_strerror_lowlevel( errnum ) != NULL ) ?                  \
        mbedtls_strerror_lowlevel( errnum ) : pNoLowLevelMbedTlsCodeStr ); \
    }
```